CHANGELOG.md

```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Disallow multiple data paths for search nodes ([#6427](https://github.com/opensearch-project/OpenSearch/pull/6427))
 
 ### Dependencies
+- Bump `org.apache.logging.log4j:log4j-core` from 2.18.0 to 2.20.0
```

Comment:
This will make the `CHANGELOG` section for `Unreleased 2.x` on `main` and `2.x` diverge.
We would also need to update the changelog on main post this PR merge or fix the changelog on main and re-run the backport workflow, which will take care of it automagically.

Reply:
Good point - I'll fix the changelog in `main` through a follow-up PR. Any concerns otherwise with merging this or #6504 in?

Reply:
PR raised to `main` - https://github.com/opensearch-project/OpenSearch/pull/6508